src/pocketmine/Player.php

```diff
@@ -1378,7 +1385,7 @@ protected function checkNearEntities($tickDiff){
 	}
 
 	protected function processMovement($tickDiff){
-		if(!$this->isAlive() or !$this->spawned or $this->newPosition === null or $this->teleportPosition !== null or $this->isSleeping()){
```

Comment:
I contest the decision that this is not needed. This will allow hacked clients to move around while dead. It also doesn't make sense to allow this regardless.